packages/react-components/react-avatar/src/components/Avatar/useAvatarStyles.ts

```diff
@@ -23,38 +23,81 @@ const animations = {
   nullEasing: tokens.curveLinear,
 };
 
-const useStyles = makeStyles({
-  root: {
-    display: 'inline-block',
-    flexShrink: 0,
-    position: 'relative',
-    verticalAlign: 'middle',
-    ...shorthands.borderRadius(tokens.borderRadiusCircular),
-    fontFamily: tokens.fontFamilyBase,
-    fontWeight: tokens.fontWeightSemibold,
-  },
+const avatarBorderRadius = '--fui-Avatar__borderRadius';
```

Comment:
I don't think it really adds any complexity and based on the data we have

1. additional class names on a DOM node lead to a linear increase in style recalc time
2. local CSS vars are really cheap

this change is probably worth it just from a style recalc point of view. Taking into account that `shorthands.borderRadius()` must construct an object only to turn around and immediately destructure it, using shorthands is almost certainly slower when it comes to JS run time as well.

Of course we're talking about a very small number of things running once per component so we're definitely in micro-optimization territory.

Since this is an implementation detail and can be changed later I'll roll this back for now.

Reply:
(Unresolving for visibility of my reply, but please re-resolve)

As far as I know, the shorthands destructuring happens in a pre-processing step by the griffel babel plugin, so that shouldn't impact runtime. If it does have a runtime impact, then we should look into changing shorthands in general, because we use it everywhere.

My main reason for my objection is that this is complicating the default and most common case of a round avatar, in exchange for a small perf win on the uncommon case of a square avatar.

In reality, the perf difference is likely negligible. But that's more reason to not introduce more code complexity for it :)

Reply:
> As far as I know, the shorthands destructuring happens in a pre-processing step by the griffel babel plugin, so that shouldn't impact runtime.

Not be default, unfortunately, and I'm learning it can be difficult to enable the Webpack plugin in large, existing applications.

Reply:
@spmonahan have you tried enabling the babel plugin instead of the webpack one? I find that it might be easier to do since it is easier to scope for large, existing applications. Allowing you to enable incrementally and, once you do that for the whole application, then allowing you to easily switch to the webpack plugin.

Reply:
First off, I was mistaken and Fluent does have the Babel plugin enabled for its build so we're good there.

@khmakoto, the [docs state the Babel plugin is for libraries and the Webpack loader is for applications](https://github.com/microsoft/griffel/tree/main/packages/webpack-loader#when-to-use-it) so that's what I've gone with.